### DISCUSSION TITLE
HTTP 1.1 Connectivity API - Source (aka Polling)

### DISCUSSION THREAD

**Discussion by jonaswre** [eclipse-ditto/ditto#1541] [Q&A]
I have read the whole documentation twice. 

Maybe I've missed something obvious, but in many scenarios its necessary to poll devices/gateways/services for data. 

Is there any build in way to achieve this? 
I would prefer not have to run my own service that polls http endpoints and sends the result to a message broker.

Thanks alot.

**Comment by thjaeckle** [comment]
The managed HTTP connection does not support polling HTTP endpoints, that's why you didn't find it documented.

That use case was not important to anyone until now.

Interesting idea to configure polling HTTP apis via http connection sources, that makes sense and also payload mapping could be applied.
Did not think about that before, I always thought a Http connection source could be a custom Http endpoint in Ditto, accepting arbitrary data via a POST.
But for that also no concrete demand existed.

If you would like to create an issue out of this discussion, please go ahead.
We would surely accept a contribution enabling this for Http connections.

**Reply by jonaswre** [reply]
Thanks for the response.

I'll create an issue out of this. 

Can you point me in the right direction? 

Is there a scheduler in the system that could make the polling work correctly? 

Infinite loop with a sleep is probably not desirable.

**Reply by thjaeckle** [reply]
Provided the pointers where to start in the created issue: #1542
